src/Fields/fieldvector.jl

```diff
@@ -1,22 +1,92 @@
 import BlockArrays
 
+
+"""
+    FieldVector
+
+A `FieldVector` is a wrapper around one or more `Field`s that acts like vector
+of the underlying arrays.
+
+It is similar in spirit to [`ArrayPartition` from
+RecursiveArrayTools.jl](https://github.com/SciML/RecursiveArrayTools.jl#arraypartition),
+but allows referring to fields by name.
+
+# Constructors
+
+    FieldVector(;name1=field1, name2=field2, ...)
+
+Construct a `FieldVector`, wrapping `field1, field2, ...` using the names
+`name1, name2, ...`.
+"""
 struct FieldVector{T, M} <: BlockArrays.AbstractBlockVector{T}
     values::M
 end
 FieldVector{T}(values::M) where {T, M} = FieldVector{T, M}(values)
 
+
+"""
+    ScalarWrapper(val) <: AbstractArray{T,0}
+
+This is a wrapper around scalar values that allows them to be mutated as part of
+a FieldVector. A call `getproperty` on a `FieldVector` with this component will
+return a scalar, instead of the boxed object.
+"""
+mutable struct ScalarWrapper{T} <: AbstractArray{T, 0}
```

Comment:
Does Ref not currently work?  it seems to provide the same basic interface

Reply:
Ah is it because all `FieldVector` backing types need to be an `AbstractArray` subtype?

Reply:
I tried using `Ref` originally, but it required too many special cases (e.g. `similar(::Ref)` isn't defined, and linear indexing doesn't work). In the end it was simpler to just define our own.